### PR TITLE
gdk-pixbuf: bump deps and several fixes

### PR DESCRIPTION
**Pull request by valgur**
This PR contains several fixes and bumping dependencies. Among them:

* Solved pkg-config variables changes introduced by Conan >= 2.0.8 (https://github.com/conan-io/conan/pull/14233)
* Recovered macOS platform. Workarounds for macOS and `shared=True`:
  *  Avoid generating gdk-pixbuf/loaders.cache (meson `custom_target` is failing due to lack of environment variables).
  * Using Conan libgettext instead of system one. Related comments: https://github.com/conan-io/conan-center-index/pull/16266#issuecomment-1501097433
* Solved static linking with libtiff and libpng (reported upstream https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/159)

Related failing PRs: #21126.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Comment by RubenRBS** [comment]
I'ce taken the opportunity to include another PR on top of this one as it needed this fix to pass, hopefully it does not cause any issues for you :)

**Comment by RubenRBS** [comment]
Reverting as it currently does not build for v2, let's have this one go thru and get merged and then we can take care of re-enabling mac in the other PR :)

**Review by RubenRBS — APPROVED** [review]

**Review by franramirez688 — CHANGES_REQUESTED** [review]
As far as I can see, this is not reported upstream, is it? I think that we should start reporting it before merging this. Wdyt @uilianries?

**Review by bdal-jsinge — COMMENTED** [review]
Not directly related to your changes, but I was trying to create a conan package for openslide too and got stuck on gdk-pixbuf recipe.

I think line 205 "gdk_pixbuf_binarydir": "${libdir1}/gdk-pixbuf-2.0/2.10" contains a typo, should say ${libdir}. For me the typo resulted in not being able to use the gdk-pixbuf package from another meson package.

**Comment by franramirez688** [comment]
> Not directly related to your changes, but I was trying to create a conan package for openslide too and got stuck on gdk-pixbuf recipe.
> 
> I think line 205 "gdk_pixbuf_binarydir": "${libdir1}/gdk-pixbuf-2.0/2.10" contains a typo, should say ${libdir}. For me the typo resulted in not being able to use the gdk-pixbuf package from another meson package.

You're right. Since Conan 2.0.8, `PkgConfigDeps` is not using `1` as the first index:

```diff
diff --git a/recipes/gdk-pixbuf/all/conanfile.py b/recipes/gdk-pixbuf/all/conanfile.py
index 4f44d79742..1ffbfc55c7 100644
--- a/recipes/gdk-pixbuf/all/conanfile.py
+++ b/recipes/gdk-pixbuf/all/conanfile.py
@@ -12,7 +12,7 @@ from conan.tools.scm import Version

 import os

-required_conan_version = ">=1.56.0 <2 || >=2.0.6"
+required_conan_version = ">=1.56.0 <2 || >=2.0.8"


 class GdkPixbufConan(ConanFile):
@@ -195,10 +195,12 @@ class GdkPixbufConan(ConanFile):
             self.cpp_info.exelinkflags = ldflags
             self.cpp_info.sharedlinkflags = ldflags

+        # Related to https://github.com/conan-io/conan/pull/14233
+        libdir_variable = "libdir1" if Version(conan_version) < "2.0" else "libdir"
         pkgconfig_variables = {
             "bindir": "${prefix}/bin",
             "gdk_pixbuf_binary_version": "2.10.0",
-            "gdk_pixbuf_binarydir": "${libdir1}/gdk-pixbuf-2.0/2.10",
+            "gdk_pixbuf_binarydir": "${%s}/gdk-pixbuf-2.0/2.10" % libdir_variable,
             "gdk_pixbuf_moduledir": "${gdk_pixbuf_binarydir}/loaders",
             "gdk_pixbuf_cache_file": "${gdk_pixbuf_binarydir}/loaders.cache",
             "gdk_pixbuf_csource": "${bindir}/gdk-pixbuf-csource",
```
@valgur could you add those changes too? Otherwise, we could open another PR to solve that.

**Comment by bdal-jsinge** [comment]
> You're right. Since Conan 2.0.8, `PkgConfigDeps` is not using `1` as the first index

Oh, it's not a typo but a version incompatibility. Thanks for explaining!

**Comment by franramirez688** [comment]
Hi @valgur 

I just added a couple of changes, including the macOS validation. I'm curious why it was there, so let's see if it passes now or if that validation could be more limited.

**Review by jwillikers — APPROVED** [review]

**Comment by valgur** [comment]
@franramirez688 Looks like `glib` cannot find a `libintl` shared library from `libgettext` on macOS. I'll try to reproduce it locally.

File: recipes/gdk-pixbuf/all/conanfile.py [hunk 1]
```diff
@@ -195,10 +212,13 @@ def package_info(self):
             self.cpp_info.exelinkflags = ldflags
             self.cpp_info.sharedlinkflags = ldflags
 
+        # Breaking change since Conan >= 2.0.8
+        # Related to https://github.com/conan-io/conan/pull/14233
+        libdir_variable = "libdir1" if Version(conan_version) < "2.0" else "libdir1"
```
Comment (uilianries) [inline recipes/gdk-pixbuf/all/conanfile.py:217]: Am i missing something here? `libdir1` is equals to `libdir1`

Comment (franramirez688) [reply recipes/gdk-pixbuf/all/conanfile.py:217]: My fault 😭 
```suggestion
        libdir_variable = "libdir1" if Version(conan_version) < "2.0" else "libdir"
```

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by franramirez688** [comment]
Sorry for moving this to DRAFT... I'd like to see if we could simplify the tons of patches applied before merging it.

File: recipes/gdk-pixbuf/all/conanfile.py [hunk 2]
```diff
@@ -149,22 +146,57 @@ def generate(self):
...
         apply_conandata_patches(self)
 
         meson_build = os.path.join(self.source_folder, "meson.build")
         replace_in_file(self, meson_build, "subdir('tests')", "#subdir('tests')")
         replace_in_file(self, meson_build, "subdir('thumbnailer')", "#subdir('thumbnailer')")
         replace_in_file(self, meson_build,
-                              "gmodule_dep.get_variable(pkgconfig: 'gmodule_supported')" if Version(self.version) >= "2.42.6"
-                              else "gmodule_dep.get_pkgconfig_variable('gmodule_supported')", "'true'")
+                        "gmodule_dep.get_variable(pkgconfig: 'gmodule_supported')" if Version(self.version) >= "2.42.6"
+                        else "gmodule_dep.get_pkgconfig_variable('gmodule_supported')", "'true'")
```
Comment (ericLemanissier) [inline recipes/gdk-pixbuf/all/conanfile.py:183]:
```suggestion
                        "gmodule_dep.get_variable(pkgconfig: 'gmodule_supported')",
                        "'true'")
```

File: recipes/gdk-pixbuf/all/conanfile.py [hunk 3]
```diff
@@ -149,22 +146,57 @@ def generate(self):
...
         # workaround https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/issues/203
         if Version(self.version) >= "2.42.6":
             replace_in_file(self, os.path.join(self.source_folder, "build-aux", "post-install.py"),
                                   "close_fds=True", "close_fds=(sys.platform != 'win32')")
```
Comment (ericLemanissier) [inline recipes/gdk-pixbuf/all/conanfile.py:187]:
```suggestion
        replace_in_file(self, os.path.join(self.source_folder, "build-aux", "post-install.py"),
                                       "close_fds=True", "close_fds=(sys.platform != 'win32')")
```

**Comment by valgur** [comment]
@franramirez688 What's the status of this PR? I assume something is still blocking it?

**Comment by franramirez688** [comment]
> @franramirez688 What's the status of this PR? I assume something is still blocking it?

@valgur I'd like to talk to the team about the possibility of simplifying all the patches that we're using in this recipe. I think I can give it another try these coming days 😁 Thanks for the reminder!

**Comment by uilianries** [comment]
Is it possible to drop the version `gdk-pixbuf/2.42.6`? I see several conditions related to that version. I also found `gtk` package using that version, but we could update it as well.

**Comment by franramirez688** [comment]
> Is it possible to drop the version gdk-pixbuf/2.42.6? I see several conditions related to that version. I also found gtk package using that version, but we could update it as well.

@uilianries `gtk` recipe uses that one as far as I can see.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**